src/Core/AdminConsole/OrganizationFeatures/Policies/IPolicyValidator.cs

```diff
@@ -41,3 +41,67 @@ public interface IPolicyValidator
     /// <param name="currentPolicy">The current policy, if any</param>
     public Task OnSaveSideEffectsAsync(PolicyUpdate policyUpdate, Policy? currentPolicy);
 }
+
+public interface IPolicyUpdateEvent
+{
+    /// <summary>
+    /// The PolicyType that this definition relates to.
+    /// </summary>
+    public PolicyType Type { get; }
+}
+
+public interface IOnPolicyPostSaveEvent : IPolicyUpdateEvent
+{
+    public Task ExecutePostUpsertSideEffectAsync(SavePolicyModel policyRequest,
+        Policy postUpdatedPolicy, Policy? previousPolicyState);
+}
+
+public interface IOnPolicyPreSaveEvent : IPolicyUpdateEvent
+{
+    public Task ExecutePreUpsertSideEffectAsync(PolicyUpdate policyUpdate, Policy? currentPolicy);
+}
+
+public interface IEnforceDependentPoliciesEvent : IPolicyUpdateEvent
+{
+    /// <summary>
+    /// PolicyTypes that must be enabled before this policy can be enabled, if any.
+    /// These dependencies will be checked when this policy is enabled and when any required policy is disabled.
+    /// </summary>
+    public IEnumerable<PolicyType> RequiredPolicies { get; }
+}
+
+public interface IPolicyValidationEvent : IPolicyUpdateEvent
+{
+    public Task<string> ValidateAsync(PolicyUpdate policyUpdate, Policy? currentPolicy);
+}
+
+public interface IPolicyEventHandlerFactory
```

Comment:
Just a small factory to easily get the policy’s implementation of that interface. 

I tried creating a default one to avoid passing around `null`, but I ran into a few problems and it didn’t provide much value, so I decided not to go through with it. It’s something we can revisit later though.